pkg/endpoint/bpf.go

```diff
@@ -1110,26 +1108,17 @@ func (e *Endpoint) deletePolicyKey(keyToDelete policy.Key, incremental bool, had
 		return false
 	}
 
-	var entry policy.MapStateEntry
-	var ok bool
-	if entry, ok = e.realizedPolicy.PolicyMapState[keyToDelete]; ok && entry.ProxyPort != 0 && hadProxy != nil {
-		*hadProxy = true
-	}
-
-	// Operation was successful, remove from realized state.
-	delete(e.realizedPolicy.PolicyMapState, keyToDelete)
 	e.updatePolicyMapPressureMetric()
```

Comment:
The `e.updatePolicyMapPressureMetric()` call is intended to update the metrics for policy map usage whenever the contents change. I think this call should also move to the caller to ensure that it is correctly updated whenever the map is modified.

Ditto for `addBPFPolicyKey()`.

Reply:
BPF policy map is still updated in this function (now called `deleteBPFPolicyKey()`), the caller updates the endpoint's realized policy map. Maybe the intent of the metric is to indicate the pressure on the BPF maps instead of the Go maps? If that is the case then the metric update is in the right place :-)